### ISSUE TITLE
Double render issue

### ISSUE THREAD

**Issue by allthesignals** [dknutsen/ember-needs-async#49]
I'm seeing the following template trigger the `find-record` task _twice_:

```handlebars
<NeedsAsync
  @needs={{find-record 'lot' model.id}}
  as |data-task|
>
  <data-task.loaded
    data-task-was-successful
    as |model|
  >
   Rendered
  </data-task.loaded>
</NeedsAsync>
```

Network requests:
![image](https://user-images.githubusercontent.com/5004319/60282338-b7768f80-98d4-11e9-82cb-c5f7ed1b8b7c.png)

It's only on subsequent renders (you will notice in my screenshot that there are 3 network requests - one for when the _route_ initially loads, two requests for only _1_ route change.

However, when I don't pass a block to the component, it behaviors correctly:

```handlebars
<NeedsAsync
  @needs={{find-record 'lot' model.id}}
  as |data-task|
/>
```

Any ideas?

**Comment by dknutsen** [comment]
Very interesting...  thanks for the report!

🤔 theoretically this should only happen if the entire template that this is located in is getting re-rendered or if of the args passed into the find-record helper is changing. If not then it definitely seems like there's a bug somewhere. Have you seen this anywhere else or is it relegated to just this one place? I'll do some testing and see what I can come up with...

**Comment by dknutsen** [comment]
@allthesignals have you discovered anything more about this? I haven't been able to reproduce in some really basic tests but I haven't dug in in-depth yet.

**Comment by allthesignals** [comment]
Hey @dknutsen, sorry I haven't gotten back to you.

I think this has to do with some unintended consequences from using contextual components and the `didReceiveAttrs` hook. The gist of it is that I was yielding something that was updating a property unexpectedly. I got around the issue by avoiding that particular component hook and using a computed property instead.

Sorry for the red herring – it was hard to isolate where the problem was coming from. Closing for now, but I do think there can be some confusion using didReceiveAttrs and contextual components...
